src/decompiler-ast.inl

```diff
@@ -89,18 +90,20 @@ struct AST {
     if (f) mc.EndFunc();
   }
 
-  void NewNode(NodeType ntype, ExprType etype, const Expr* e, Index nargs) {
-    assert(stack.size() >= nargs);
+  Node& NewNode(NodeType ntype, ExprType etype, const Expr* e, Index nargs,
```

Comment:
Oo that got weird. What is dip? (It seems to be an "n from the end" offset, but that's not completely obvious). This returning a reference to the newly inserted node feels convenient, but odd.
Partly the naming of "NewNode" makes it feel extra odd to return a value that is already in-place. I'd expect something more like `exp_stack.push_back(NewNode(...));`. It having returned void in the past made it more clear that that wasn't how "new" was being used, but now seeing `NewNode(...).foo = ...;` looks surprising.

Reply:
`dip` in stack languages is exactly that: do an operation as if N from the top of the stack is the actual top of the stack :)

I'll rename it `InsertNode` which is closer to what it does. What it actually does is `InsertNewNodeAndTakeExistingNodesOnTheStackAsChildren` but I can put that in a comment.